app/tests/src/actions/transaction-actions.js

```diff
@@ -28,22 +28,99 @@ it('Calculate fee - shelley', async () => {
   mockNet.mockRewardHistory()
   mockNet.mockPoolRecommendation()
+})
 
+const loadTestWallet = async () => {
   await action.loadWallet(state, {
     cryptoProviderType: CryptoProviderType.WALLET_SECRET,
-    walletSecretDef: await mnemonicToWalletSecretDef(walletSettings[0].secret),
+    walletSecretDef: await mnemonicToWalletSecretDef(walletSettings.Shelley15Word.secret),
   })
+}
+
+const sendAdaTxSettings = {
+  donation: {
+    state: {
+      sendAddress: {
+        fieldValue:
+          'addr1qjag9rgwe04haycr283datdrjv3mlttalc2waz34xcct0g4uvf6gdg3dpwrsne4uqng3y47ugp2pp5dvuq0jqlperwj83r4pwxvwuxsgds90s0',
+      },
+      sendAmount: {fieldValue: 1, coins: 1000000},
+      donationAmount: {fieldValue: 5, coins: 5000000},
+    },
+    sendTransactionSummary: {
+      amount: 1000000,
+      donation: 5000000,
+      fee: 183419,
+      plan: {},
+      tab: 'send',
+      deposit: 0,
+    },
+  },
+}
 
-  state.sendAddress.fieldValue =
-    'addr1qjag9rgwe04haycr283datdrjv3mlttalc2waz34xcct0g4uvf6gdg3dpwrsne4uqng3y47ugp2pp5dvuq0jqlperwj83r4pwxvwuxsgds90s0'
-  state.sendAmount.fieldValue = 1
-  state.sendAmount.coins = 1000000
-  state.donationAmount.fieldValue = 5
-  state.donationAmount.coins = 5000000
+const delegationSettings = {
+  delegation: {
+    state: {
+      shelleyDelegation: {
+        selectedPool: {
+          poolHash: '04c60c78417132a195cbb74975346462410f72612952a7c4ade7e438',
+        },
+      },
+    },
+    sendTransactionSummary: {
+      amount: 0,
+      donation: 0,
+      fee: 193878,
+      plan: {},
+      tab: 'stake',
+      deposit: 2000000,
+    },
+  },
+}
 
-  await action.calculateFee()
+const withdrawalSettings = {
+  rewardWithdrawal: {
+    state: {},
+    sendTransactionSummary: {
+      amount: 0,
+      donation: 0,
+      fee: 185089,
+      plan: {},
+      tab: 'withdraw',
+      deposit: 0,
+    },
+  },
+}
 
-  assert.equal(state.transactionFee, 183419)
+describe('Send ADA fee calculation', () => {
+  Object.entries(sendAdaTxSettings).forEach(([name, setting]) =>
+    it(`should calculate fee for tx with ${name}`, async () => {
+      await loadTestWallet()
+      setStateFn(state, setting.state)
+      await action.calculateFee()
+      assert.deepEqual(state.sendTransactionSummary.fee, setting.sendTransactionSummary.fee)
+    })
+  )
+})
+
+describe('Delegation fee calculation', () => {
+  Object.entries(delegationSettings).forEach(([name, setting]) =>
+    it(`should calculate fee for tx with ${name}`, async () => {
+      await loadTestWallet()
+      setStateFn(state, setting.state)
+      await action.calculateDelegationFee()
+      assert.deepEqual(state.sendTransactionSummary.fee, setting.sendTransactionSummary.fee)
+    })
+  )
+})
 
-  mockNet.clean()
+describe('Withdrawal fee calculation', () => {
+  Object.entries(withdrawalSettings).forEach(([name, setting]) =>
+    it(`should calculate fee for tx with ${name}`, async () => {
+      await loadTestWallet()
```

Comment:
setting.state could be passed into loadTestWallet as a parameter and perform the `setStateFn` within it, removing that repetitiveness from the tests themselves

Reply:
Good idea :+1: